modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -3574,6 +3563,8 @@ else if (!spi.failureDetectionTimeoutEnabled() && (e instanceof
 
                                     int res = spi.readReceipt(sock, timeoutHelper.nextTimeoutChunk(ackTimeout0));
 
+                                    updateLastSentMessageTime();
```

Comment:
what the reason to set sent time here?

Reply:
We just sent one or several pending messages. Every message can take time, also every message checks connection. We rely on the time of last sent message. We should fixate this time when do really send message. We do it in this part of code too.